### ISSUE TITLE
[BUG/ISSUE] Restart file generated for unexpected date

### ISSUE THREAD

**Issue by MaggieMarvin** [geoschem/geos-chem#86]
**Describe the bug**
In a simulation starting 20130101 and ending 20131231, a restart file was produced for 20131201, although the restart file was requested for output at the end of the simulation. However, a HEMCO restart file was produced for the correct date. I can pick up my next simulation starting from 20131201, but this was an unexpected result and I'm not sure what's caused it.

**To Reproduce**
1. In input.geos file:
Start YYYYMMDD, hhmmss  : 20130101 000000
End   YYYYMMDD, hhmmss  : 20131231 000000
2. In HISTORY.rc file:
Restart.filename:          './GEOSChem.Restart.%y4%m2%d2_%h2%n2z.nc4',
Restart.format:             'CFIO',
Restart.frequency:       'End',
Restart.duration:          'End',
Restart.mode:              'instantaneous'
Restart.fields:              'SpeciesRst_?ALL?               ', 'GIGCchem',

**Expected behavior**
A restart file was expected to be produced at the end of the simulation (20131231), rather than at 20131201.

**Required information**
 - GEOS-Chem version you are using: 12.5.0
 - Compiler version that you are using: gfortran 4.8.5
 - netCDF and netCDF-Fortran library version: 4.3.3.1
 - Computational environment: local computer
 - Are you using "out of the box" code, or have you made modifications? out of the box

**Input and log files to attach**
For more info, see: http://wiki.geos-chem.org/Submitting_GEOS-Chem_support_requests
 - The GEOS-Chem "Classic" log file
[run.log](https://github.com/geoschem/geos-chem/files/3711638/run.log)

**Comment by yantosca** [comment]
Hi Maggie,

It looks like your input.geos file is scheduled to end on 12/31:

```
In input.geos file:
Start YYYYMMDD, hhmmss : 20130101 000000
End YYYYMMDD, hhmmss : 20131231 000000
```

Try changing the ending time to
```
End YYYYMMDD, hhmmss : 20140101 000000
```

and then you should get a restart file on 2014/01/01 at 0 GMT.

**Comment by MaggieMarvin** [comment]
Hi Bob,

Thanks! The intention was to use this as a spin-up simulation, then to start collecting boundary conditions one day before 2014 so that I can run a custom 0.25x0.3125 nested grid from 2014 on. 

However, I'm using MERRA2 meteorology for the 2013 spin-up simulation and GEOS-FP for 2014, because offline lightning emissions are not available for GEOS-FP for 2013. So I think I'll have to mash boundary conditions from MERRA2 for 00 hh on 20140101 with GEOS-FP for the other hours on that day to get a complete set of boundary conditions. But this will still require BC collection prior to 20140101, so maybe I can still leverage this output and start collecting BCs from 20131201... Is there an easier way to do all this?

Best,
Maggie

**Comment by yantosca** [comment]
One thing I just thought of... the HEMCO restart file is set to "End" in HEMCO_Diagn.rc.  That explains why you got output at the end of the run.  (You can also set it to "monthly", "daily', "hourly", or other frequencies.

The GEOS-Chem restart (and boundary conditions) are produced up to the end date in input.geos.  You could try stretching the end date to e.g.  20131231 235000.  That would probably produce the last day of boundary conditions (I think).   You would be missing some data from the last hour but that might be good enough.

**Comment by MaggieMarvin** [comment]
I tried ending the run on 20140101 000000, and this produced the GEOS and HEMCO restart files for the correct day and hour. It also produced boundary conditions for 20140101 at 00 hh only, so I think I should be able to merge this file with boundary conditions collected from 20140101 with GEOS-FP, and this will give me a complete set of boundary conditions for that day. Thanks!

**Comment by yantosca** [comment]
Glad that worked.  I'll close this issue out.  An easy way to merge data is to use xarray (or you could also use the ncrcat operator of NCO).  We have more on our [Working with netCDF data files](http://wiki.geos.chem.org/Working_with_netCDF_data_files) page on the wiki.

**Comment by neyranaz** [comment]
Hello Bob, 
Geoschem 12.7.0

We are experiencing the same issue that @MaggieMarvin was experienced. Specifically, we are running an aerosol only nested grid run from 2014-2016. In our observations we came up with a potential root to the error. It seems that when naming the restart file, (after it was generated), GC is taking the month from the end of the segment, and combining it with the day from the first segment. So for Maggie's example it would be taking  Beginning 201301**01** , End  2013**12**31, therefore creating the geoschem restartfile  20131201.   

We observed this issue when performing a `geosfp_025x3125_aerosol_na` nested grid run.   
The segments that we ran were: 

20140102-20140211 
20140211-20140412 
20140412-20140621
20140621-20140829 

Our restart files are 

GEOSChem.Restart.20140101_0000z.nc4
GEOSChem.Restart.20140202_0000z.nc4
GEOSChem.Restart.20140411_0000z.nc
GEOSChem.Restart.20140612_0000z.nc4
GEOSChem.Restart.20140821_0000z.nc4

Our HEMCO.restart files are: 

HEMCO_restart.201402110000.nc
HEMCO_restart.201404120000.nc
HEMCO_restart.201406210000.nc
HEMCO_restart.201408290000.nc
 
While the name of the geoschem.restart files seems to be wrong, the run segment inside the file seem to be right, we did an `ncdump -h ` and the segments shown did match those segments of our run. 

Also, it seems that when GC reads the restart files, it also takes the switched dates, so it lets GC start correctly. We confirmed this by looking at our `run.log` file. There, we observed that GC, reads the file `GEOSChem.Restart.20140612_0000z.nc4 ` for the run segment  `20140621-20140829 ` in `input.geos` . Also, after it reads the file, it does show the correct segment inside GC. 

Although we believe that the fact that GC is pulling the restart file with the dates switched too, might not cause an error (for our set up). We are concerned that this would be the case. So we wanted to know if you would suggest to do the run all over again. Or to run it using 01. 

While this seems not to be an issue for our purposes, as we configured our GC restart files HISTORY.rc, to be produced at the `End` of every segment, it might be an issue for other projects in which the generation of the restart file was done on a daily, weekly, monthly, or any other basis. This would be because, when restarting a new segment, GC would pull the file with the switched dates, and it wouldn't bring an error (maybe), because the segment date inside the geoschem.restart file would apparently be correct, but the file would not be. So if the restart is set to Daily for example, the next segment would initialize incorrectly or generate a [timestamp](https://github.com/geoschem/geos-chem/issues/248#issuecomment-680211127) error depending on the segment inside the file. 

Finally, we believe the error should be rooted  here: 

in HISTORY.rc 
`Restart.filename: './GEOSChem.Restart.%y4%m2%d2_%h2%n2z.nc4',`

In HEMCO_Config.rc
`./GEOSChem.Restart.$YYYY$MM$DD_$HH$MNz.nc4`

This might be very wrong, but we wanted to ask you about this, as this could really bias some estimations.

**Comment by yantosca** [comment]
Thanks for writing @neyranaz.  See: issues https://github.com/geoschem/geos-chem/issues/305 and https://github.com/geoschem/geos-chem/issues/269, which were resolved in GEOS-Chem 12.8.1.

**Comment by neyranaz** [comment]
Hello @Yantosca , Thanks a lot for your prompt reply. We coincidentally had already seen in detail the issues that you mention. We even applied the commit from issue #305 based on a previous suggestion that you gave. However, we are not sure if issue #269, and issue [#248](https://github.com/geoschem/geos-chem/issues/248) correct this pattern that seems to apply uniquely to restart file generation, is this the case?    

It is important to mention that we tried to use version 12.8.1 and we tried newer versions too. However, we weren't able to do this as offline_aerosol (5x4 and nested) runs were not being possible  due to issue [#354](https://github.com/geoschem/geos-chem/issues/354 )
